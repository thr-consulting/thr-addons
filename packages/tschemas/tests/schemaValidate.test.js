import yup from 'yup';
import TSchemas from '../src/index';

const obj = {
	anumber: 5,
	astring: 'hello world',
};

const schema = yup.object({
	anumber: yup.number().required(),
	astring: yup.string().required(),
});

it('should validate', async () => {
	expect.assertions(1);
	expect(await TSchemas.schemaValidate(schema, obj)).toBeNull();
});
it('should fail validation', async () => {
	expect.assertions(1);
	expect(await TSchemas.schemaValidate(schema, {anumber: 'this is not a number'})).toMatchSnapshot();
});
