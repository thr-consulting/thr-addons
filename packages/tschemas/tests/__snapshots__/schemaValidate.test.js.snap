// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fail validation 1`] = `
Object {
  "errors": Array [
    "anumber must be a \`number\` type, but the final value was: \`\\"this is not a number\\"\`.",
  ],
  "inner": Array [],
  "message": "anumber must be a \`number\` type, but the final value was: \`\\"this is not a number\\"\`.",
  "name": "ValidationError",
  "params": Object {
    "label": undefined,
    "originalValue": "this is not a number",
    "path": "anumber",
    "type": "number",
    "value": "this is not a number",
  },
  "path": "anumber",
  "type": "typeError",
}
`;
